### ISSUE TITLE
Hapi - authentication is not working.

### ISSUE THREAD

**Issue by fenrirx22** [slanatech/swagger-stats#109]
Couldn't find out any issue with this. I'm trying to hide my swagger-stats behind the authentication. Everything works except the authentication. I've set the authentication param to true and provide the handler for the callback and nothing happens. I'm out of the ideas.
I'm using hapi-swagger 13.0.2, "swagger-stats": "^0.95.17".

```
return server.register({
    plugin: swStats.getHapiPlugin,
    options: {
      name: "Testing metrics",
      version: "1.0",
      authentication: true,
      onAuthenticate: (req, username, password) => {
        console.log("auth!");
        return (
          username === config.METRICS_USERNAME &&
          password === config.METRICS_PASS
        );
      }
    }
  });
```

**Comment by fenrirx22** [comment]
I've done a little investigation. Seems like in the `swsHapi` file, routes are added for stats/metrics/ui/ux but the handler are using just processor functions to obtain those. The whole auth callback is called on `swsInterface` functions that wraps up the processor ones.

**Comment by fenrirx22** [comment]
I've created a PR to fix this. https://github.com/slanatech/swagger-stats/pull/110

Btw, there are some tests on master that are not working ( supertest with elasticsearch - 500_elastic.js ). I didn't repair them because the test settings are pretty hard to understand, would take too much time  and honestly I do not agree with this kind of setup (like one test depend on the execution of the previous one ).

**Comment by fenrirx22** [comment]
I've created a PR to fix this. #110

Btw, there are some tests on master that are not working ( supertest with elasticsearch - 500_elastic.js ). I didn't repair them because the test settings are pretty hard to understand, would take too much time and honestly I do not agree with this kind of setup (like one test depend on the execution of the previous one ).

**Comment by sv2** [comment]
Addressed v0.99.1
